include/net/wifi_mgmt.h

```diff
@@ -109,18 +109,22 @@ struct wifi_status {
 };
 
 #include <net/net_if.h>
+#include <net/ethernet.h>
 
 typedef void (*scan_result_cb_t)(struct net_if *iface, int status,
 				 struct wifi_scan_result *entry);
 
-struct net_wifi_mgmt_offload {
+struct net_wifi_mgmt {
 	/**
 	 * Mandatory to get in first position.
 	 * A network device should indeed provide a pointer on such
 	 * net_if_api structure. So we make current structure pointer
 	 * that can be casted to a net_if_api structure pointer.
 	 */
-	struct net_if_api iface_api;
+    union {
```

Comment:
In #35719 I have subbmitted the direction which we've more or less agreed on when we've discusses WiFi MGMT rework to support partially offloaded drivers (which make use of the native net stack). Please have a look at https://github.com/rlubos/zephyr/commits/wifi/mgmt-integration-rework, it would be good to straighten this up once and for goodinstead of introducing workarounds like this union.

Reply:
Yes, your approach is way nicer than mine. How do we proceed on this? Do you want to submit this change first or should i cherry-pick your changes into my pr?

Reply:
We're planning to submit a separate PR with those changes. But feel free to cherry-pick them temporarily into your PR.